### DISCUSSION TITLE
Nextjs App Environemnt variable handling

### DISCUSSION THREAD

**Discussion by rbeach4** [nrwl/nx#16242] [Feature Requests]
Hello,

I have been using NX for awhile and still have not found a good way to handle environment variables at runtime, rather than build time. This is a breaking issue for a lot of people. It would be nice to get this fixed and working ASAP.

Ricky Beach

**Comment by rbeach4** [comment]
This is specifically for Nextjs app within NX

**Comment by marko-hologram** [comment]
I believe this is more tied to how Next.js works. Next.js replaces `process.env.*` with actual values during build time. There is a way to use runtime variables with Next.js, but that is considered legacy and disables some optimizations and features that Next.js has https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration

As far as I know this has nothing to do with NX and it's just the way Next.js works by default. I believe NX cannot do anything about this.
